src/clp_logging/handlers.py

```diff
@@ -792,3 +800,168 @@ def __init__(
+        loglevel_timeout: Optional[CLPLogLevelTimeout] = None,
+    ) -> None:
+        super().__init__()
+
+        self._enable_compression: bool = enable_compression
+        self._tz: Optional[str] = timezone
+        self._loglevel_timeout: Optional[CLPLogLevelTimeout] = loglevel_timeout
+        self._serializer: Optional[Serializer] = None
+        self._formatter: Optional[logging.Formatter] = None
+        self._ostream: IO[bytes] = stream
+
+        self._auto_generated_kv_pairs_buf: AutoGeneratedKeyValuePairsBuffer = (
+            AutoGeneratedKeyValuePairsBuffer()
+        )
+
+        self._init_new_serializer(stream)
+
+    # override
+    def setFormatter(self, fmt: Optional[logging.Formatter]) -> None:
+        if fmt is None:
+            return
+        warnings.warn(
+            f"Formatter is currently not supported in the current {self.__class__.__name__}",
+            category=RuntimeWarning,
+        )
+        self._formatter = fmt
+
+    # override
+    def emit(self, record: logging.LogRecord) -> None:
+        """
+        Overrides `logging.Handler.emit` to ensure `logging.Handler.handleError`
+        is always called and avoid requiring a `logging.LogRecord` to call
+        internal writing functions.
+
+        :param record: The log event to serialize.
+        """
+        try:
+            self._write(record)
+        except Exception:
+            self.handleError(record)
+
+    # Added to `logging.StreamHandler` in python 3.7
+    # override
+    def setStream(self, stream: IO[bytes]) -> Optional[IO[bytes]]:
+        if stream is self._ostream:
+            return None
+        old_stream: IO[bytes] = self._ostream
+        self._ostream = stream
+        # TODO: The following call will close the old stream. However, `logging.StreamHandler`'s
+        # implementation will only flush the stream but leave it opened. To support this behaviour,
+        # we need `clp_ffi_py.ir.Serializer` to allow closing the serializer without closing the
+        # underlying output stream.
+        self._init_new_serializer(stream)
+        return old_stream
+
+    # override
+    def close(self) -> None:
+        if self._serializer is None:
+            return
+        if self._loglevel_timeout:
+            self._loglevel_timeout.timeout()
+        # NOTE: Closing the serializer will ensure that any buffered results are flushed and the
+        # underlying output stream is properly closed.
+        self._serializer.close()
+        self._serializer = None
+        super().close()
+
+    def _is_closed(self) -> bool:
+        return self._serializer is None
+
+    def _init_new_serializer(self, stream: IO[bytes]) -> None:
+        """
+        Initializes a new serializer that writes to the given stream.
+
+        :param stream: The stream that the underlying serializer will write to.
+        """
+        cctx: ZstdCompressor = ZstdCompressor()
+        self._serializer = Serializer(
```

Comment:
* Shouldn't we close the existing Serializer before replacing it?
* We should note that we close the stream in the docstring, right?